src/libfsm/print/ir.h

```diff
@@ -54,8 +54,10 @@ struct ir_error {
 struct ir_state {
 	const char *example;
 
-	fsm_end_id_t *ids; /* NULL -> 0 */
-	size_t count:31; // :31 for packing
+	struct ir_state_endids {
+		fsm_end_id_t *ids; /* NULL -> 0 */
```

Comment:
This loses the `:31 for packing`, but if the goal is to save memory per IR state, then this could instead be a separately allocated struct like `struct ir_state_endids { size_t count; fsm_end_ids_t ids[]; }` and states without endids (which will be most states) save storing the count entirely. Thoughts?

Reply:
No need, the packing isn't that important.